onto_tool/onto_tool.py

```diff
@@ -1036,6 +1041,101 @@ def __verify_shacl__(action, variables):
             exit(1)
 
 
+def __verify_tool__(action, variables, tools):
+    with tempfile.TemporaryDirectory() as tempdir:
+        tool = next((t for t in tools if t['name'] == action['tool']), None)
+        if not tool:
+            raise Exception('Missing tool ', action['tool'])
+
+        data_graph = __build_graph_from_inputs__(action, variables)
+        data_file = os.path.join(tempdir, "data.ttl")
+        data_graph.serialize(data_file, format="turtle", encoding="utf-8")
+        if "shapes" in action:
+            shape_graph = __build_graph_from_inputs__(action['shapes'], variables)
+            shape_file = os.path.join(tempdir, "shapes.ttl")
+            shape_graph.serialize(shape_file, format="turtle", encoding="utf-8")
+            logging.debug("Shape graph has %s triples", sum(1 for _ in shape_graph))
+        else:
+            shape_file = None
+        logging.debug("Data graph has %s triples", sum(1 for _ in data_graph))
+
+        arguments = []
+        if tool['type'] == 'Java':
+            arguments = ["java", "-jar", tool['jar']] + tool['arguments']
+        elif tool['type'] == 'shell':
+            arguments = tool['arguments']
+        else:
+            raise Exception('Unsupported tool type for verify', tool['type'])
+
+        outputPath, interpreted_args = __run_verify_tool__(action, variables, data_file, shape_file, arguments)
+
+        if outputPath:
+            __report_violations_from_tool__(action, outputPath, interpreted_args)
+
+
+def __run_verify_tool__(action, variables, data_file, shape_file, arguments):
+    invocation_vars = VarDict()
+    invocation_vars.update(variables)
+    invocation_vars.update({
+            'dataFile': data_file,
+            'shapeFile': shape_file
+        })
+    outputPath = None
+    if action.get('target'):
+        outputPath = action['target'].format(**variables)
+        invocation_vars.update({'outputFile': outputPath})
+    interpreted_args = [arg.format(**invocation_vars) for arg in arguments]
+    logging.debug('Running %s', interpreted_args)
+    try:
+        status = subprocess.run(interpreted_args, capture_output=True)
+    except Exception:
+        exc_type, exc_value, exc_traceback = sys.exc_info()
+        traceback.print_tb(exc_traceback, limit=30, file=sys.stdout)
+        print(exc_value, file=sys.stdout)
+        print(exc_type, file=sys.stdout)
+        sys.exit()
```

Comment:
I don't think we need to print the traceback (which should be for internal failures) - if the user provided a bad path for a tool, that should be reported as a regular issue, using `logging.error()`, we should not have `print()` statements in the code. Also, a `sys.exit(1)` should be used to indicate an error termination.